raiden/blockchain/state.py

```diff
@@ -1,47 +1,38 @@
+from typing import TYPE_CHECKING
+
 from raiden.settings import MEDIATION_FEE
 from raiden.transfer.identifiers import CanonicalIdentifier
 from raiden.transfer.state import (
     NettingChannelEndState,
     NettingChannelState,
     TransactionExecutionStatus,
 )
-from raiden.utils.typing import (
-    BlockNumber,
-    BlockTimeout,
-    Optional,
-    PaymentNetworkAddress,
-    TokenAddress,
-    TokenNetworkAddress,
-)
+from raiden.utils import typing
 
+if TYPE_CHECKING:
+    from raiden.network.proxies.payment_channel import ChannelDetails, PaymentChannel
 
-def get_channel_state(
-    token_address: TokenAddress,
-    payment_network_address: PaymentNetworkAddress,
-    token_network_address: TokenNetworkAddress,
-    reveal_timeout: BlockTimeout,
-    payment_channel_proxy,
-    opened_block_number: BlockNumber,
-):
-    # Here we have to query the latest state because if we query with an older block
-    # state (e.g. opened_block_number) the state may have been pruned which will
-    # lead to an error.
-    latest_block_hash = payment_channel_proxy.client.blockhash_from_blocknumber("latest")
-    channel_details = payment_channel_proxy.detail(latest_block_hash)
 
+def do_get_channel_state(
```

Comment:
please change to something more telling, i.e. `create_channel_state_from_blockchain_data`.

Reply:
Good idea.